EIPS/eip-5169.md

```diff
@@ -0,0 +1,191 @@
+
+1. A Utility NFT is an event ticket and the authenticated script is a JavaScript 'minidapp' which asks the user to sign a challenge message that shows ownership of the key controlling the ticket. The dapp would then render the signature as a QR code which can be scanned by a ticketing app, which could then mark the ticket as used.
+
+2. Smart Token Labs uses a framework called TokenScript; one element of which is a user interface description for contract interaction through tokens.
+Consider a simple 'mint' verb associated with an already existing NFT. The associated script can for example allow the owner to mint a derivative through a contract already holding enough ether for the minting fee, without needing to connect their wallet.
+
+3. An NFT Script which controls a Smartlock. For example consider the lock  being linked to a digital NFT twin and being controlled with the verbs "lock" and "unlock", each of which has an associated JavaScript. Each of these scripts could be executed after the user signs a challenge in a web-view. This is an off-chain example that uses on-chain assets for functionality.
+
+### Tests
+#### Test Contract
+```
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "./IERC5XX0.sol";
+contract ERC5XX0 is IERC5XX0, Ownable {
+    string[] private _scriptURI;
+    function scriptURI() external view override returns(string[] memory) {
+        return _scriptURI;
+    }
+
+    function setScriptURI(string[] memory newScriptURI) external onlyOwner override {
+        _scriptURI = newScriptURI;
+
+        emit ScriptUpdate(newScriptURI);
+    }
+}
+```
+
+#### Test case
+```
+const { expect } = require('chai');
+const { BigNumber, Wallet } = require('ethers');
+const { ethers, network, getChainId } = require('hardhat');
+
+describe('ERC5XX0', function () {
+  before(async function () {
+    this.ERC5XX0 = await ethers.getContractFactory('ERC5XX0');
+  });
+
+  beforeEach(async function () {
+    // targetNFT
+    this.erc5XX0 = await this.ERC5XX0.deploy();
+  });
+
+  it('Should set script URI', async function () {
+    const scriptURI = [
+      'uri1', 'uri2', 'uri3'
+    ];
+
+    await expect(this.erc5XX0.setScriptURI(scriptURI))
+      .emit(this.erc5XX0, 'ScriptUpdate')
+      .withArgs(scriptURI);
+    
+    const currentScriptURI = await this.erc5XX0.scriptURI();
+
+    expect(currentScriptURI.toString()).to.be.equal(scriptURI.toString());
+  });
+```
+
+### Security Considerations
+
+**When a server is involved**
+
+When the client script does not purely rely on connection to a blockchain node, but also calls server APIs,  the trustworthiness of the server API is called into question. This ERC doesn't provide the mechanism to assert the authenticity of the API access point. Instead, as long as the client script is trusted, it's assumed that it can call any server API in order to carry out token functions. This means the client script can mistrust a server API access point.
```

Comment:
```suggestion
When the client script does not purely rely on connection to a blockchain node, but also calls server APIs,  the trustworthiness of the server API is called into question. This EIP does not provide any mechanism to assert the authenticity of the API access point. Instead, as long as the client script is trusted, it's assumed that it can call any server API in order to carry out token functions. This means the client script can mistrust a server API access point.
```